src/include/btree.i

```diff
@@ -489,6 +489,8 @@ __wt_page_only_modify_set(WT_SESSION_IMPL *session, WT_PAGE *page)
     if (page->modify->page_state < WT_PAGE_DIRTY &&
       __wt_atomic_add32(&page->modify->page_state, 1) == WT_PAGE_DIRTY_FIRST) {
         __wt_cache_dirty_incr(session, page);
+        if (page->read_gen == WT_READGEN_WONT_NEED)
```

Comment:
The comment you added has a good level of information, but would be better if it didn't reference the name of the define. We try to use regular words in comments rather than variable and definition names - it forces the comment to be an abstraction rather than a statement of what the code is doing. You'll be able to find a ton of counter examples, but capturing this sort of thing is why we use the spell checker on comments in s_string.